weatherAPI/libraries/response.go

```diff
@@ -0,0 +1,152 @@
+package libraries
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/astaxie/beego/httplib"
+	"github.com/astaxie/beego/logs"
+)
+
+func GetResponse(city, country string) map[string]interface{} {
+	var jsonResponse map[string]interface{}
+	response := make(map[string]interface{})
+
+	uri := fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%v,%v&appid=1508a9a4840a5574c822d70ca2132032", city, country)
+	res := httplib.Get(uri) //get the uri sending
+	res.Debug(true)
+	str, e := res.String() //return raw response body
+	CheckErrors("Error returning the raw response: ", e)
+	err := json.Unmarshal([]byte(str), &jsonResponse)
+	CheckErrors("Error trying to unmarshal the data: ", err)
+
+	//getting the json data
+	coord := jsonResponse["coord"].(map[string]interface{})
+	wind := jsonResponse["wind"].(map[string]interface{})
+	main := jsonResponse["main"].(map[string]interface{})
+	sys := jsonResponse["sys"].(map[string]interface{})
+	clouds := jsonResponse["clouds"].(map[string]interface{})
+	//getting time
+	sunrise := time.Unix(int64(sys["sunrise"].(float64)), 0)
+	sunset := time.Unix(int64(sys["sunset"].(float64)), 0)
+	timer := time.Unix(int64(jsonResponse["dt"].(float64)), 0)
+	//getting wind values
+	b, w, wd := beaufort(wind["speed"].(float64)), wind["speed"].(float64), windir(wind["deg"].(float64))
+
+	response["location_name"] = fmt.Sprintf("%s, %s", jsonResponse["name"], sys["country"])
+
+	response["temperature"] = fmt.Sprintf("%.0f °C", main["temp"].(float64)-273.15)
+
+	response["wind"] = fmt.Sprintf("%s, %.2f m/s, %s", b, w, wd)
+
+	response["cloudiness"] = fmt.Sprintf("%s", cloudiness(clouds["all"].(float64)))
+
+	response["pressure"] = fmt.Sprintf("%.0f hpa", main["pressure"])
+
+	response["humidity"] = fmt.Sprintf("%.0f%%", main["humidity"])
+
+	response["sunrise"] = fmt.Sprintf("%02d:%02d", sunrise.Hour(), sunrise.Minute())
+
+	response["sunset"] = fmt.Sprintf("%02d:%02d", sunset.Hour(), sunset.Minute())
+
+	response["geo_coordinates"] = []float64{coord["lon"].(float64), coord["lat"].(float64)}
+
+	response["requested_time"] = fmt.Sprintf(
+		"%v",
+		time.Date(timer.Year(), timer.Month(), timer.Day(), timer.Hour(), timer.Minute(), timer.Second(), timer.Nanosecond(), timer.Location()))
```

Comment:
No sure what you're trying to do, but for time conversion you can use the standard library.